pkgs/by-name/mt/mtkclient/package.nix

```diff
@@ -0,0 +1,50 @@
+{
+  lib,
+  fetchFromGitHub,
+  python3Packages,
+}:
+
+python3Packages.buildPythonApplication {
+  pname = "mtkclient";
+  version = "2.0.1-unstable-2025-09-26";
+  pyproject = true;
+
+  src = fetchFromGitHub {
+    owner = "bkerler";
+    repo = "mtkclient";
+    rev = "399b3a1c25e73ddf4951f12efd20f7254ee04a39";
+    hash = "sha256-XNPYeVhp5P+zQdumS9IzlUd5+WebL56qcgs10WS2/LY=";
+  };
+
+  build-system = [ python3Packages.hatchling ];
+
+  dependencies = with python3Packages; [
+    colorama
+    fusepy
+    pycryptodome
+    pycryptodomex
+    pyserial
+    pyside6
+    pyusb
+    shiboken6
+  ];
+
+  pythonImportsCheck = [ "mtkclient" ];
+
+  # Note: No need to install mtkclient udev rules, 50-android.rules is covered by
+  #       systemd 258 or newer and 51-edl.rules only applies to Qualcomm (i.e. not MTK).
```

Comment:
25.05 doesn't have systemd 258 (yet?), so the appropriate solution here would be to reference `android-udev-rules`. However, I don't think backports usually make voluntary edits to the commits?

Reply:
In the grand scheme of things the comment probably doesn't matter too much. We probably aren't going to make `mtkclient` depend on `android-udev-rules` just for the release branch, so the end result for what users have to do doesn't change, they have to include that package in their configuration manually anyway.

In my opinion we could merge this into 25.05 as-is just fine. That said, I *do* have my reservations about upstream flakes that rely on the downstream nixpkgs definition, so I don't quite agree with the motivation for backporting in itself.